### PR TITLE
Fixed bug: setting broken flag too early. Breaks too early data transmission

### LINKED ISSUE #1234: files not received completely[recvfile,sendfile]
I'm use the example program to send and recv file(examples/sendfile.cpp,recvfile.cpp).but the file not received completely.
the udp packet not lost.I had checked with netstat -us. no new packet receive errors.

Udp:
    774152 packets received
    4056 packets to unknown port received.
    16013 packet receive errors
    794625 packets sent
    16013 receive buffer errors
    0 send buffer errors

the log below:
[root@localhost srt]# ./sendfile
15:08:46.216922/sendfile D:SRT.c: generateSocketID: : @96419073
15:08:46.217295/sendfile D:SRT.c: CHANNEL: Bound to local address: 0.0.0.0:9000
server is ready at port: 9000
15:08:57.414025/SRT:RcvQ:w1.N:SRT.c: PASSING request from: 127.0.0.1:39399 to agent:96419073
15:08:57.414688/SRT:RcvQ:w1.N:SRT.c: Listener managed the connection request from: 127.0.0.1:39399 result:waveahand
15:08:57.414830/SRT:RcvQ:w1.N:SRT.c: PASSING request from: 127.0.0.1:39399 to agent:96419073
15:08:57.414876/SRT:RcvQ:w1 D:SRT.c: generateSocketID: : @96419072
15:08:57.415302/SRT:RcvQ:w1.N:SRT.c: HSREQ/rcv: cmd=1(HSREQ) len=12 vers=0x10401 opts=0xe4 delay=0
15:08:57.415382/SRT:RcvQ:w1.N:SRT.c: listen ret: -1 - conclusion
15:08:57.415395/SRT:RcvQ:w1.N:SRT.c: Listener managed the connection request from: 127.0.0.1:39399 result:waveahand
new connection: 127.0.0.1:39399
speed = 1417.01Mbits/sec
loss = 0pkt (0%)

-----------------------------------------------------------------------------------------

[root@localhost srt]# ./recvfile 127.0.0.1 9000 /root/go1.14.1.linux-amd64.tar.gz go.tar.gz
15:08:57.413273/recvfile D:SRT.c: generateSocketID: : @171403877
15:08:57.417616/recvfile.N:SRT.c: @171403877:Connection established to: 127.0.0.1:9000
speed = 3.56142Mbits/sec
loss = 0pkt (0%)

-----------------------------------------------------------------------------------------

[root@localhost ~]# ll go1.14.1.linux-amd64.tar.gz 
-rw-r--r--. 1 root root 123631885 4月   8 03:46 go1.14.1.linux-amd64.tar.gz
[root@localhost srt]# ll go.tar.gz 
-rw-r--r--. 1 root root 110711328 4月   9 15:08 go.tar.gz

with tar xzvf go.tar.gz error
gzip: stdin: unexpected end of file
go/src/cmd/vendor/golang.org/x/arch/arm/armasm/tables.go

I had tested for many time, for small file size,it will work fine. so what's wrong?

### PR DESCRIPTION
**Pull request by ethouris**
Fixes #1234 

The problem was that newly introduced function `makeClosed` was used in more places than it was necessary and therefore it was setting the `m_bBroken` flag too early. This caused inability to dispatch incoming packets and rejected any other transmission, which then led to closing the connection before the file can be completely transmitted.

### CODE REVIEW

**Comment by ethouris** [comment]
If we are going to make this change, I'd research a little bit deeper, if setting `m_bBroken` is really required before calling `CUDT::close()`. Because if so, it would be simpler split into smaller functions.

File: srtcore/api.cpp [hunk 1]
```diff
@@ -124,7 +124,10 @@ SRT_SOCKSTATUS CUDTSocket::getStatus()
     return m_Status;
 }
 
-void CUDTSocket::makeClosed()
+/// This makes the socket no longer capable of performing any transmission
+/// operation, but continues to be responsive in the connection in order
+/// to finish sending the data that were scheduled for sending so far.
+void CUDTSocket::makeShutdown()
```
Comment (maxsharabayko) [inline srtcore/api.cpp:130]: I would suggest placing this description at the function declaration (api.h) instead of the function definition.

Comment (maxsharabayko) [reply srtcore/api.cpp:130]: Also, I find your earlier description more clear.
@stevomatthews Could you please advise on this?

> Shutdown makes it no longer possible to perform send/recv operation on the socket, but REMAINING DATA from the previous requests continue to be performed until the queues are empty or until the linger expires.

Comment (ethouris) [reply srtcore/api.cpp]: You copied the text together with my mistake :D
